kernel/src/scan/mod.rs

```diff
@@ -195,24 +218,23 @@ impl Scan {
         Ok(scan_action_iter(
             engine,
             log_iter,
-            &self.read_schema,
+            &self.logical_schema,
             &self.predicate,
         ))
     }
 
     /// Get global state that is valid for the entire scan. This is somewhat expensive so should
     /// only be called once per scan.
     pub fn global_scan_state(&self) -> GlobalScanState {
-        let partition_columns = &self.snapshot.metadata().partition_columns;
-        let (_all_fields, read_fields, _have_partition_cols) =
-            get_state_info(self.schema().as_ref(), partition_columns);
-        let read_schema = StructType::new(read_fields);
-        let logical_schema = self.schema().as_ref().clone();
+        let partition_columns = self.snapshot.metadata().partition_columns.clone();
+        let logical_schema = self.logical_schema.as_ref().clone();
+        let read_schema = self.physical_schema.as_ref().clone();
         GlobalScanState {
             table_root: self.snapshot.table_root.to_string(),
-            partition_columns: partition_columns.clone(),
+            partition_columns,
             logical_schema,
             read_schema,
+            column_mapping_mode: self.column_mapping_mode.clone(),
         }
     }
```

Comment:
nit: we mix two "styles" here, assignment to a variable, and assignment directly to the struct field. Should we settle on one?

Reply:
hah, i was looking at this and wondering if I should harmonize it. since you noticed it too, i will :)

Reply:
I usually go with whatever produces fewer lines of code... we _might_ be able to fit everything here by direct field assignment, if `partition_columns` isn't too long?